test/integration/recover_update_local_connection_test.go

```diff
@@ -0,0 +1,50 @@
+// +build recover
+
+package nsmd_integration_tests
+
+import (
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+
+	"github.com/networkservicemesh/networkservicemesh/test/kubetest"
+)
+
+func TestUpdateConnectionOnNSEChange(t *testing.T) {
+	RegisterTestingT(t)
+
+	if testing.Short() {
+		t.Skip("Skip, please run without -short")
+		return
+	}
+
+	k8s, err := kubetest.NewK8s(true)
+	Expect(err).To(BeNil())
+	defer k8s.Cleanup()
+
+	nodesConf, err := kubetest.SetupNodes(k8s, 2, defaultTimeout)
+	Expect(err).To(BeNil())
+	defer kubetest.FailLogger(k8s, nodesConf, t)
+
+	nse1 := kubetest.DeployICMP(k8s, nodesConf[0].Node, "icmp-responder-nse-1", defaultTimeout)
+
+	nsc := kubetest.DeployMonitoringNSC(k8s, nodesConf[0].Node, "monitoring-nsc", defaultTimeout)
+	k8s.WaitLogsContains(nsc, "monitoring-nsc", "Monitor started", defaultTimeout)
+
+	time.Sleep(5 * time.Second) // we need to be sure that NSC is updated only with the initial connection
```

Comment:
Looks like at that point we already know that logs contain "Monitor started" which is written only in case of receiving initial connection. So why do we need that Sleep?

Reply:
Oh seems like to ensure that there are no more received events